File: examples/shopflow/sdks/c/src/services.c
#include "services.h"
#include <stdlib.h>
#include <string.h>

app_response_t* app_dummy_json_get_categories(app_client_t* client, const app_request_opts_t* opts) {
    app_request_opts_t local_opts = {0};
    if (opts) local_opts = *opts;
    local_opts.method = "GET";
    local_opts.endpoint = "products/categories";
    return app_client_request(client, &local_opts);
}

app_response_t* app_dummy_json_get_category_list(app_client_t* client, const app_request_opts_t* opts) {
    app_request_opts_t local_opts = {0};
    if (opts) local_opts = *opts;
    local_opts.method = "GET";
    local_opts.endpoint = "products/category-list";
    return app_client_request(client, &local_opts);
}

app_response_t* app_dummy_json_get_current_user(app_client_t* client, const app_request_opts_t* opts) {
    app_request_opts_t local_opts = {0};
    if (opts) local_opts = *opts;
    local_opts.method = "GET";
    local_opts.endpoint = "auth/me";
    return app_client_request(client, &local_opts);
}

app_response_t* app_dummy_json_get_product_by_id(app_client_t* client, const app_request_opts_t* opts) {
    app_request_opts_t local_opts = {0};
    if (opts) local_opts = *opts;
    local_opts.method = "GET";
    local_opts.endpoint = "products/{id}";
    return app_client_request(client, &local_opts);
}

app_response_t* app_dummy_json_get_products(app_client_t* client, const app_request_opts_t* opts) {
    app_request_opts_t local_opts = {0};
    if (opts) local_opts = *opts;
    local_opts.method = "GET";
    local_opts.endpoint = "products";
    return app_client_request(client, &local_opts);
}

app_response_t* app_dummy_json_get_products_by_category(app_client_t* client, const app_request_opts_t* opts) {
    app_request_opts_t local_opts = {0};
    if (opts) local_opts = *opts;
    local_opts.method = "GET";
    local_opts.endpoint = "products/category/{category}";
    return app_client_request(client, &local_opts);
}

app_response_t* app_dummy_json_login(app_client_t* client, const app_request_opts_t* opts) {
    app_request_opts_t local_opts = {0};
    if (opts) local_opts = *opts;
    local_opts.method = "POST";
    local_opts.endpoint = "auth/login";
    return app_client_request(client, &local_opts);
}

app_response_t* app_dummy_json_search_products(app_client_t* client, const app_request_opts_t* opts) {
    app_request_opts_t local_opts = {0};
    if (opts) local_opts = *opts;
    local_opts.method = "GET";
    local_opts.endpoint = "products/search";
    return app_client_request(client, &local_opts);
}

File: examples/shopflow/sdks/rust/src/services.rs
use crate::client::Client;
use reqwest::header::HeaderMap;
use std::collections::HashMap;

/// Service for DummyJsonService containing 8 endpoints
pub struct DummyJsonService;

impl DummyJsonService {
    /// GET products/categories
    pub async fn get_categories(
        client: &Client,
    ) -> Result<reqwest::Response, reqwest::Error> {
        let pp_arg = None;
        client.send_request(
            reqwest::Method::GET,
            "products/categories",
            pp_arg,
            None,
            None,
            None,
        ).await
    }

    /// GET products/category-list
    pub async fn get_category_list(
        client: &Client,
    ) -> Result<reqwest::Response, reqwest::Error> {
        let pp_arg = None;
        client.send_request(
            reqwest::Method::GET,
            "products/category-list",
            pp_arg,
            None,
            None,
            None,
        ).await
    }

    /// GET auth/me
    pub async fn get_current_user(
        client: &Client,
    ) -> Result<reqwest::Response, reqwest::Error> {
        let pp_arg = None;
        client.send_request(
            reqwest::Method::GET,
            "auth/me",
            pp_arg,
            None,
            None,
            None,
        ).await
    }

    /// GET products/{id}
    pub async fn get_product_by_id(
        client: &Client,
        id: &str,
    ) -> Result<reqwest::Response, reqwest::Error> {
        let mut path_params = HashMap::new();
        path_params.insert("id", id);
        let pp_arg = Some(&path_params);
        client.send_request(
            reqwest::Method::GET,
            "products/{id}",
            pp_arg,
            None,
            None,
            None,
        ).await
    }

    /// GET products
    pub async fn get_products(
        client: &Client,
        query_params: Option<&HashMap<&str, &str>>,
    ) -> Result<reqwest::Response, reqwest::Error> {
        let pp_arg = None;
        client.send_request(
            reqwest::Method::GET,
            "products",
            pp_arg,
            query_params,
            None,
            None,
        ).await
    }

    /// GET products/category/{category}
    pub async fn get_products_by_category(
        client: &Client,
        category: &str,
        query_params: Option<&HashMap<&str, &str>>,
    ) -> Result<reqwest::Response, reqwest::Error> {
        let mut path_params = HashMap::new();
        path_params.insert("category", category);
        let pp_arg = Some(&path_params);
        client.send_request(
            reqwest::Method::GET,
            "products/category/{category}",
            pp_arg,
            query_params,
            None,
            None,
        ).await
    }

    /// POST auth/login
    pub async fn login(
        client: &Client,
        payload: Option<&serde_json::Value>,
    ) -> Result<reqwest::Response, reqwest::Error> {
        let pp_arg = None;
        client.send_request(
            reqwest::Method::POST,
            "auth/login",
            pp_arg,
            None,
            None,
            payload,
        ).await
    }

    /// GET products/search
    pub async fn search_products(
        client: &Client,
        query_params: Option<&HashMap<&str, &str>>,
    ) -> Result<reqwest::Response, reqwest::Error> {
        let pp_arg = None;
        client.send_request(
            reqwest::Method::GET,
            "products/search",
            pp_arg,
            query_params,
            None,
            None,
        ).await
    }

}
